rfcs/0003-mutations.md

```diff
@@ -0,0 +1,317 @@
+
+## Terminology
+
+* _Mutations_: Collection of mutations.  
+* _Mutation_: A GraphQL mutation.  
+* _Mutations Schema_: A GraphQL schema that defines a `type Mutation` that contains all mutations. Additionally, this schema can define other types to be used by the mutations, such as `input` and `interface` types.  
+* _Mutations Manifest_: A YAML manifest file that is used to add mutations to an existing subgraph manifest.  
+* _Mutation Resolvers_: Code module that contains all resolvers.  
+* _Resolver_: Function that is used to execute a mutation.  
+* _Mutation State_: The state of a mutation being executed. It's passed to the resolver through the mutation context.  
+* _Mutation Context_: A context object that's created for every mutation that's executed. It's passed as an argument to the resolver.  
+* _Config_: Collection of config properties required by the mutation resolvers.  
+* _Config Property_: A single property within the config (ex: ipfs, ethereum, etc).  
+* _Config Generator_: A function that takes a config value, and returns a config property. For example, "localhost:5001" as a config value gets turned into a new IPFS client by the config generator.
+* _Config Value_: An initialization value that's passed into the config generator. This config value is provided by the dApp developer.
+* _Optimistic Response_: A response given to the dApp that predicts what the outcome of the mutation's execution will be. If it is incorrect, it will be overwritten with the actual result.  
+
+## Detailed Design
+
+The sections below illustrate how a developer would add mutations to an existing subgraph, and then add those mutations to a dApp.
+
+### Mutations Manifest
+
+The subgraph manifest (`subgraph.yaml`) now has an extra property named `mutations` which is the mutations manifest.
+
+`subgraph.yaml`
+```yaml
+specVersion: ...
+...
+mutations:
+  repository: https://npmjs.com/package/...
+  schema:
+    file: ./mutations/schema.graphql
+  resolvers:
+    apiVersion: 0.0.1
+    kind: javascript/es5
+    file: ./mutations/index.js
+dataSources: ...
+...
+```
+
+Alternatively, the mutation manifest can be external like so:  
+`subgraph.yaml`
+```yaml
+specVersion: ...
+...
+mutations:
+  file: ./mutations/mutations.yaml
+dataSources: ...
+...
+```
+`mutations/mutations.yaml`
+```yaml
+repository: https://npmjs.com/package/...
+schema:
+  file: ./schema.graphql
+resolvers:
+  apiVersion: 0.0.1
+  kind: javascript/es5
+  file: ./index.js
+```
+
+### Mutations Schema
+
+The mutations schema defines all of the mutations in our subgraph. The mutations schema is a super-set of the subgraph's schema. For example, starting from a base subgraph schema:  
```

Comment:
How about `The mutations schema is a super-set of the subgraph's schema` -> `The mutations schema builds on the subgraph schema by allowing to use types from the subgraph schema, as well as defining new types that are used only in the context of mutations.`?

Reply:
Because it's not a superset.